### DISCUSSION TITLE
Redirect to previous on unguarded routes

### DISCUSSION THREAD

**Discussion by hsks** [nuxt-community/auth-module#1570] [Ideas & Feature Requests]
Hi everyone,
I know that the module supports redirection after logging in to guarded routes using the `home` and `rewriteRedirect` options. Is there a similar approach offered by the module for scenarios where we might want to trigger authentication explicitly on a public route?
For example, let's say on clicking a button `Log In` on an unguarded route called `public-route` we trigger the authentication using **loginWith("strategy")**. Looks like the redirection doesn't work out of the box for such scenarios. It only works for guarded routes?

**Comment by reinoldus** [comment]
I very much support this feature. What I am doing to workaround this is the following:

Disable all auth-module redirects:

    redirect: {
      home: false,
      callback: '/authorize',
      rewriteRedirects: false,
      fullPathRedirect: false,
    },

And handle everything yourself in your callback hook:

```
  async mounted() {
    await this.$auth.fetchUser()
    if (this.$auth.$storage.getCookie('redirect_after_login') !== undefined) {
      const val = this.$auth.$storage.getCookie('redirect_after_login')
      this.$auth.$storage.setCookie('redirect_after_login', undefined, false)
      await this.$router.push(val)
    } else if (this.$auth.$storage.getCookie('checkout_signup') !== undefined) {
      const val = this.$auth.$storage.getCookie('checkout_signup')
      this.$auth.$storage.setCookie('checkout_signup', undefined, false)
      await this.$router.push(val)
    } else {
      await this.$router.push('/user/')
    }
  },
```

Important is this line:  `await this.$auth.fetchUser()` otherwise the user might no be logged in. I assume auth-module is hooking into something, somewhere and this hook might not be able to complete in some cases. Downside is that the user will get fetched twice and login takes a bit longer

It might not be the wisest idea to use this.$auth.$storage.get/setCookie not sure about that

**Comment by hsks** [comment]
I'll definitely give this a try.
We're currently leveraging the browser's own local storage to keep a track of where to redirect the user to after the callback is successfully mounted. It's very similar to your approach I'd say, the only difference being the use of a local storage.
Fingers crossed, auth module may start supporting this out of the box 🤞

**Comment by nhanco** [comment]
> I very much support this feature. What I am doing to workaround this is the following:
> 
> Disable all auth-module redirects:
> 
> ```
> redirect: {
>   home: false,
>   callback: '/authorize',
>   rewriteRedirects: false,
>   fullPathRedirect: false,
> },
> ```
> 
> And handle everything yourself in your callback hook:
> 
> ```
>   async mounted() {
>     await this.$auth.fetchUser()
>     if (this.$auth.$storage.getCookie('redirect_after_login') !== undefined) {
>       const val = this.$auth.$storage.getCookie('redirect_after_login')
>       this.$auth.$storage.setCookie('redirect_after_login', undefined, false)
>       await this.$router.push(val)
>     } else if (this.$auth.$storage.getCookie('checkout_signup') !== undefined) {
>       const val = this.$auth.$storage.getCookie('checkout_signup')
>       this.$auth.$storage.setCookie('checkout_signup', undefined, false)
>       await this.$router.push(val)
>     } else {
>       await this.$router.push('/user/')
>     }
>   },
> ```
> 
> Important is this line: `await this.$auth.fetchUser()` otherwise the user might no be logged in. I assume auth-module is hooking into something, somewhere and this hook might not be able to complete in some cases. Downside is that the user will get fetched twice and login takes a bit longer
> 
> It might not be the wisest idea to use this.$auth.$storage.get/setCookie not sure about that

I was try this but when redirected u got no user's info need a refresh to got the data.
